src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -659,7 +659,14 @@ public static function delete( $thread_id = 0, $user_id = 0 ) {
 		$bp = buddypress();
 
 		// Get the message ids in order to pass to the action.
-		$message_ids = $wpdb->get_col( $wpdb->prepare( "SELECT id FROM {$bp->messages->table_name_messages} WHERE thread_id = %d", $thread_id ) ); // WPCS: db call ok. // WPCS: cache ok.
+		$messages    = BP_Messages_Message::get(
+			array(
+				'fields'          => 'ids',
+				'include_threads' => array( $thread_id ),
+			)
+		);
+
+		$message_ids = ( isset( $messages['messages'][0] ) ) ? $messages['messages'][0] : array();
```

Comment:
@janakkaneriya007 Should be like below. Old query returning array if ids i guess and you returning First id of array.
```suggestion
		$message_ids = ( isset( $messages['messages'] ) ) ? $messages['messages'] : array();
```